### ISSUE TITLE
False positive `W0612` / 'unused variable' when using same name for exception in except block

### ISSUE THREAD

**Issue by DervishD** [pylint-dev/pylint#10426]
### Bug description

```python
With the minimal code below a `W0612` warning is produced for the second except, but it goes if a different variable name is used instead of `exc`. For some reason, using the same variable name in both except blocks confuses the linter and causes the false positive.


try:
    pass
except TypeError as exc:
    print(exc)
except ValueError as exc:
    print(exc)
```

### Configuration

```ini
[MASTER]
jobs=0
unsafe-load-any-extension=yes

[MESSAGES CONTROL]
enable=all
disable=I,too-many-lines

[REPORTS]
#reports=yes
msg-template='{path}:{line}:{column}: {msg_id} ({symbol}) {msg}'

[BASIC]
good-names=i,j,c,_
include-naming-hint=yes

[FORMAT]
max-line-length=120

[VARIABLES]
allow-global-unused-variables=no
```

### Command used

```shell
I'm using Visual Studio Code Pylance, so the command is like:
`python.exe -m pylint --reports=n --output-format=json --clear-cache-post-run=y --from-stdin minimal.py`
```

### Pylint output

```python
{
        "type": "warning",
        "module": "erase",
        "obj": "",
        "line": 5,
        "column": 0,
        "endLine": 6,
        "endColumn": 14,
        "path": "C:\\Users\\rauln\\Desktop\\sysbin\\erase.py",
        "symbol": "unused-variable",
        "message": "Unused variable 'exc'",
        "message-id": "W0612"
    }
```

### Expected behavior

No `W0612` warning issued for that code.

### Pylint version

```shell
pylint 3.3.4
astroid 3.3.8
Python 3.13.2 (tags/v3.13.2:4f8bb39, Feb  4 2025, 15:23:48) [MSC v.1942 64 bit (AMD64)]
```

### OS / Environment

Windows 11 Home 24H2
Visual Studio Code 1.101.0
Pylance  2025.6.1

### Additional dependencies

```python

```

**Comment by zenlyj** [comment]
Thanks for the report, there seems to be an issue with the `allow-global-unused-variables` configuration. The warning is raised only when the flag is set to "no".

**Comment by DervishD** [comment]
Thanks @zenlyj ! Yes, I reported another similar bug report which only fired when that flag is set to "no". I have that in my config file because for my projects it makes sense to be warned about unused globals, even though I admit things can get a bit noisy on test suites, for example.

Anyway, if you need me to perform any test or provide more information, just tell.

**Comment by DervishD** [comment]
Thanks a lot!
